tests/func/test_external_repo.py

```diff
@@ -26,3 +27,11 @@ def test_external_repo(erepo):
             assert path_isin(repo.cache.local.cache_dir, repo.root_dir)
 
         assert mock.call_count == 1
+
+
+def test_external_repo_import_without_remote(erepo, dvc_repo):
```

Comment:
But `erepo` has default remote setup, so your test doesn't work at all. You need to remove default remote setting logic from erepo fixture, as it won't be needed, since you've introduced it in _external_repo in this PR.